catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -0,0 +1,507 @@
+
+            query.setFields("*", RELEVANCE_SORT_FIELD);
+
+            if (Result.RELEVANCE.equals(sortProperty)) {
+                query.addSort(RELEVANCE_SORT_FIELD, order);
+            } else if (Result.DISTANCE.equals(sortProperty)) {
+                addDistanceSort(query, GEOMETRY_SORT_FIELD, order, solrFilterDelegate);
+            } else if (sortProperty.equals(Result.TEMPORAL)) {
+                query.addSort(resolver.getSortKey(resolver.getField(Metacard.EFFECTIVE,
+                        AttributeType.AttributeFormat.DATE,
+                        false)), order);
+            } else {
+                List<String> resolvedProperties = resolver.getAnonymousField(sortProperty);
+
+                if (!resolvedProperties.isEmpty()) {
+                    for (String sortField : resolvedProperties) {
+                        if (sortField.endsWith(SchemaFields.GEO_SUFFIX)) {
+                            addDistanceSort(query,
+                                    resolver.getSortKey(sortField),
+                                    order,
+                                    solrFilterDelegate);
+                        } else if (!(sortField.endsWith(SchemaFields.BINARY_SUFFIX)
+                                || sortField.endsWith(SchemaFields.OBJECT_SUFFIX))) {
+                            query.addSort(resolver.getSortKey(sortField), order);
+                        }
+                    }
+                } else {
+                    LOGGER.debug(
+                            "No schema field was found for sort property [{}]. No sort field was added to the query.",
+                            sortProperty);
+                }
+
+            }
+
+        }
+        return resolver.getSortKey(sortProperty);
+    }
+
+    private ResultImpl createResult(SolrDocument doc) throws MetacardCreationException {
+        ResultImpl result = new ResultImpl(createMetacard(doc));
+
+        if (doc.get(RELEVANCE_SORT_FIELD) != null) {
+            result.setRelevanceScore(((Float) (doc.get(RELEVANCE_SORT_FIELD))).doubleValue());
+        }
+
+        if (doc.get(DISTANCE_SORT_FIELD) != null) {
+            Object distance = doc.getFieldValue(DISTANCE_SORT_FIELD);
+
+            if (distance != null) {
+                LOGGER.debug("Distance returned from Solr [{}]", distance);
+                double convertedDistance = new Distance(Double.valueOf(distance.toString()),
+                        Distance.LinearUnit.KILOMETER).getAs(Distance.LinearUnit.METER);
+
+                result.setDistanceInMeters(convertedDistance);
+            }
+        }
+
+        return result;
+    }
+
+    private Double degreesToMeters(double distance) {
+        return new Distance(DistanceUtils.degrees2Dist(distance,
+                DistanceUtils.EARTH_MEAN_RADIUS_KM),
+                Distance.LinearUnit.KILOMETER).getAs(Distance.LinearUnit.METER);
+    }
+
+    public MetacardImpl createMetacard(SolrDocument doc) throws MetacardCreationException {
```

Comment:
Looks like the `CacheSolrMetacardClient` needs to call `setId` on the metacard which requires the impl. See https://github.com/codice/ddf/blob/3fc83610ed9b85a23005cb075b843d0b10f0473f/catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/CacheSolrMetacardClient.java#L58